text/0122-sdk-hub-scope-merge.md

```diff
@@ -0,0 +1,458 @@
+
+![image](./0122-images/scope-inheritance.png)
+
+### How to store/reference scopes
+
+Similar to how we currently store the current hub, we also need to store the current scope. The exact mechanism to do that is up to the concrete SDK/language. In environments with multiple execution contexts (or thread locals), we should attach the current scope to the execution context (or thread local). In other environments, it may be attached to some global (e.g. in Browser JS). This RFC does not propose to change this, so generally the current scope may be stored the same was as the current hub used to be.
+
+In addition to the current scope, the current [Global Scope](#global-scope) and [Isolation Scope](#isolation-scope) (see below for an in-depth explanation on what these are) also need to be stored. These should be stored _next to_ the current scope. If you have executution context, you may store/access them like this:
+
+```js
+const executionContext1 = getExecutionContext();
+const { scope, globalScope, isolationScope } = executionContext1.getScopes();
+executionContext1.setScopes({ scope, globalScope, isolationScope });
+```
+
+In environments where you don't have execution contexts, you may store them in a global place. The exact mechanism to do this is up to the concrete SDK/language.
+
+For each execution context, there will _always_ be exactly one current, global & isolation scope. When the execution context is forked, these are forked along.
+
+### Global Scope
+
+In addition to the current scope, there will also be a new special scope, the **Global Scope**.
+
+When you call `Sentry.init()`, you will immediately have a current scope - an initial scope.
+In the current/previous system, if you want to set global data (e.g. tags that should apply to _all_ events),
+you have to do this right after calling `Sentry.init()`, on the initial scope, so this data will be correctly inherited by all downstream scopes:
+
+```js
+Sentry.init();
+
+// Scope is the initial scope - the scope right after calling init()
+cont scope = Sentry.getCurrentHub().getCurrentScope();
+scope.setTag('revision', 'abc');
+```
+
+This becomes especially tricky if you want to set global scope data based on async processes, or in different files.
+The global scope should fix this by making it very clear how to setup data/processors that should apply to _all_ events, no matter where you define them:
+
+```js
+Sentry.init();
+setupGlobalData();
+
+// Somewhere else
+async function setupGlobalData() {
+  const revision = await getRevision();
+  // No need to worry or care about where this is called, this will _always_ work!
+  Sentry.getGlobalScope().setTag("revision", "abc");
+}
+```
+
+The global scope is _not_ the initial scope, but a separate, special scope. The global scope is different for each client, in order to avoid polluting this on accident when working with multiple completely separate clients.
```

Comment:
I am confused as to why the global scope is different for each client. If I got this right, do we need some bookkeeping of global scopes for clients? How is that going to work?
I would expect that the global scope is global 😄. It applies to all events.